recipes/views.py

```diff
@@ -39,13 +39,10 @@ def edit(request, recipe_id):
 
 def detail(request, recipe_id):
     recipe = get_object_or_404(Recipe, pk=recipe_id)
-    return render(request, 'recipes/detail.html', {'recipe': recipe})
-
-@login_required
-def ingredients(request, recipe_id):
-    recipe = get_object_or_404(Recipe, pk=recipe_id)
-    context = {'ingredients': recipe.ingredients.all(), 'recipe_id': recipe_id}
-    return render(request, 'recipes/ingredients.html', context)
+    ingredients = recipe.ingredients.all()
+    instructions = recipe.instructions.all()
+    return render(request, 'recipes/detail.html',
+                  {'recipe': recipe, 'ingredients': ingredients, 'instructions': instructions})
```

Comment:
Explicitly getting the ingredients and instructions is not really necessary here. In your template you could simply reference recipe.ingredients.all(), etc. 

Having the ability to do it like this is still good though since sometimes your models don't happen to have all the info you need on them.